### ISSUE TITLE
cisco amp and malwarebye detecting as malware

### ISSUE THREAD

**Issue by tmottabr** [malxau/yori#22]
ysetup and the update cab are being detected as trojan by several engines, so is being blocked by my corporate anti-virus (cisco amp and malwarebyte).

https://www.virustotal.com/gui/file/568d3e98510528d8fa1db59a0c21963933d0b7719610d623bbdad34098ad6956/detection

**Comment by malxau** [comment]
Do you have any suggestions on actions I can take?

From some superficial Googling:
- Cisco accept reports of false positives from customers, and I am not a customer
- Malwarebytes suggest posting to their forum at https://forums.malwarebytes.org/index.php?/forum/42-file-detections/ , although my attempts to create an account there were rejected as spam.  The rejection includes a link to a helpdesk to talk to about overactive rejection, and that link is broken.

Somebody emailed me about this earlier, so to paraphrase part of my reply to them:
- Note that antivirus vendors have a very difficult technical problem of trying to distinguish software that downloads software that users want and software that downloads software that users don't want.  Arguably this is impossible and they're left drawing inferences from weakly correlated indicators.  If any indicator became a strongly correlated indicator, malware would try to adopt that behavior to avoid detection.
- Antivirus vendors have a business that depends on indicating to customers that they are being protected.  A vendor that has false positives is telling customers they are protected.  A vendor with false negatives is regarded as a bad security product.  So these companies are encouraged to block overactively and wait for complaints, and in some cases, don't bother accepting complaints.
- Having determined something is malware, the security product will not observe that software executing successfully with user consent, so deprives itself of any indicators that would contradict the conclusion it has made. 

Unfortunately for antivirus vendors that I don't have contacts in, there's not much I can do other than point out that they are falsely detecting things and people who choose antivirus products should bear that in mind when deciding which products to use and deploy.

**Comment by tmottabr** [comment]
not sure what else you can do other than report the false positives. But i think you should be aware of this problem, this i way i reported here. Also so other that face this problem can find this information as well.

i was able to create an account in malwarebyte and reported this false positive, the problem with malwarebyte is not the executable but the domain you host it from. it was flagged likely because ysetup.exe is being reported as trojan by some engines.

https://forums.malwarebytes.com/topic/256521-httpwwwmalsmithnet/

**Comment by malxau** [comment]
As of this writing, it looks like the only remaining engine detecting ysetup.exe is TrendMicro-HouseCall.  I have an email thread with them, but it looks to be going in circles, unfortunately.

**Comment by tmottabr** [comment]
malwarebyte is no longer blocking the domain as well.

thanks.

**Comment by jcmpgit** [comment]
My org uses Zscaler ZIA as a web proxy, this i think uses Virus total which is still showing as being blocked for one scan.
https://www.virustotal.com/gui/url/c15de5c4a0020b7336a598e5ba88c238dd01457674e44940bfe78224a94ff826/detection

**Comment by malxau** [comment]
As of this writing trend is no longer flagging the file, and the only remaining thing complaining about the URL is clean-mx.  If anyone can see a way to report false positives to them, please let me know.

**Comment by aleaksunder** [comment]
It seems that "false positives" situation is unsolvable without extra effort...
You coult read other talented developer struggling here: [http://blog.nirsoft.net/2009/05/17/antivirus-companies-cause-a-big-headache-to-small-developers/](url)
And of course nobody cares! So I have digged a little into the problem, and it seems the only answer for this is to sign all executables with your own certificate, which can not be obtained for free... the only appropriate option for start is here:
[https://www.certum.eu/en/cert_offer_en_open_source_cs](url)

Well in a nutshell... this certificate must be bought each year! This is not silver-bullet, and this will not protect your executable from being detected as viruses, but... the logic here is simple for my understanding: the longer exists your software signed with your certificate ( without direct complains from users of course ) the greater reputation your certificate gains from antivirus vendors, and of course in start you will have to write letters to them asking to clean "false positive" alert, but as time go on than less you will have to do so... in theory of course :)

Of course nobody force you to buy certificate. We all know that this is free project and no one owes nothing to nobody, but it seems to me that this is the only "less headache" way to... and maybe one "Donate" button will provide some funds to bring this to life! At least in amount of donation you could see how others are intrested in development and maybe if there will be enough funds you could afford the most expensive way get rid of a headache as described here: https://www.certum.eu/en/cert_offer_microsoft_code_signing/

By the way... this could help to protect your software and users from MITM attacks partially described in #21 

I have tried a lot of versions of CL.exe provided my Microsoft trying a "silly way" to get rid of "false positives" =) nothing seriously helps, the only thing I have noticed that a lot of "false positives" ( about 10-20 per building all source according to VirusTotal ) is only for x86 build and only for the network related stuff in all your source, so not being able to build `yoripkg` and linked to it stuff becomes almost clean binaries ( 1-3 false positives )... like amd64 build, which in full build variant without any exceptions seems by VirusTotal to be almost perfect

I have some issues with builds on different versions of compilers but will open another issue to describe it later

**Comment by malxau** [comment]
As of now, the binaries scan cleanly and the URL is detected by one vendor who is not interested in accepting that it's a false positive (which also means that vendor is not taken seriously.)  I'm closing this now because as of this instant there's nothing left to do, although it's inevitable that false positive reports will emerge from time to time.
